### PR TITLE
preparing to HepMC3

### PR DESCRIPTION
**Pull request by mkirsano**
#### PR description:

The event structure is to be changed because HepMC3 requires a correct tree to build a record. This is the main content of this PR (99%). Changed also some test configurations in Pythia8Interface/test, used by very limited number of people, intended to see the difference of tauola tau decays from pythia8 tau decays. They will be needed for further tests (essentially technical, without physics).

### CODE REVIEW

File: GeneratorInterface/Pythia8Interface/test/Py8_Z2tautau_cfg.py [hunk 1]
```diff
@@ -34,33 +34,29 @@
...
+				     ### Py6 mstp(82)=4 (MPI model) does NOT have anything similar in Py8, 
 				     ### because it actually means a new model, and in Py8 old models went away 
 				     ### - only new remain
-				     'MultipleInteractions:pT0Ref=1.8387',    # parp(82)=1.8387 - pt cutoff for MI
-				     'MultipleInteractions:ecmRef=1960.',     # parp(89)=1960.
-				     'MultipleInteractions:coreFraction=0.5', # D; parp(83)=0.5
-				     'MultipleInteractions:coreRadius=0.4',   # D; parp(84)=0.4
-				     ### parp(85) & parp(86) are N/A in Py8
-				     'MultipleInteractions:ecmPow=0.16',      # parp(90)=0.16
+				     #'MultipleInteractions:pT0Ref=1.8387',    # parp(82)=1.8387 - pt cutoff for MI
```
Comment (perrotta) [inline GeneratorInterface/Pythia8Interface/test/Py8_Z2tautau_cfg.py:41]: What is the purpose of keeping these parameters commented out?

**Comment by perrotta** [comment]
@agrohsje do you have further comments?

**Comment by qliphy** [comment]
ping @cms-sw/generators-l2

**Comment by agrohsje** [comment]
I am generally fine with the PR. I just have 2 comments:
1.) If the root line is not used I would use something that cannot be confused with a proper value, i.e. anything but 14 or 13 TeV that doesn't look like a random entry. 
2.) For the testing file. It seems the file has a long history. The current version turns settings on, that are on per default and it doesn't specify any UE settings, despite claiming so. I would suggest to clean this further up, use CP2 or whatever tune. 
As Mikhail can sign for GEN I leave it to him to decide.

**Comment by Saptaparna** [comment]
@mkirsano and @agrohsje: anything the L2s should do? We just discussed this PR at ORP and wanted to follow up. Thanks!

**Comment by mkirsano** [comment]
It would be good to thoroughly test Py8EGun once this is merged.
Sorry, I forgot to connect to ORP

**Comment by agrohsje** [comment]
please test

**Comment by agrohsje** [comment]
+generators

**Comment by perrotta** [comment]
@mkirsano , before accepting this for merge, could you please confirm that the modified configurations for `py8UESettings` and `py8ProcessSettings` (with a few parameters now unset) are actually desired? 
Could you please then also shortly address those modification in the PR descriptions?

**Comment by mkirsano** [comment]
The configurations changed are intended to compare tau decays in pythia and in tauola. I think that actually during the last years they were used only by pythia8 integrators, namely by me, and mainly to check that tauola works in principle.

**Comment by perrotta** [comment]
+1

**Pull request merged**